rotkehlchen/chain/manager.py

```diff
@@ -803,6 +846,58 @@ def modify_btc_account(
         # At the very end add/remove it from the accounts
         getattr(self.accounts.btc, append_or_remove)(account)
 
+    def modify_bch_account(
+            self,
+            account: BTCAddress,
+            append_or_remove: str,
+            add_or_sub: AddOrSub,
+            already_queried_balance: Optional[FVal] = None,
+    ) -> None:
+        """Either appends or removes a BCH acccount.
+
+        If already_queried_balance is not None then instead of querying the balance
+        of the account we can use the already queried one.
+
+        Call with 'append', operator.add to add the account
+        Call with 'remove', operator.sub to remove the account
+
+        May raise:
+        - RemotError if there is a problem querying blockchain.info or cryptocompare
+        """
+        bch_usd_price = Inquirer().find_usd_price(A_BCH)
+        remove_with_populated_balance = (
+            append_or_remove == 'remove' and len(self.balances.bch) != 0
+        )
+        # Query the balance of the account except for the case when it's removed
+        # and there is no other account in the balances
+        if append_or_remove == 'append' or remove_with_populated_balance:
```

Comment:
not something to do in your PR ... but damn this chain/manager.py has grown into a monster. There is a lot of potential for refactoring into smaller re-usable modular components.